src/ORM/Search/SearchContext.php

```diff
@@ -176,40 +205,117 @@ public function getQuery($searchParams, $sort = false, $limit = false, $existing
             $query = $query->limit($limit);
         }
 
-        /** @var DataList $query */
-        $query = $query->sort($sort);
-        $this->setSearchParams($searchParams);
+        return $query->sort($sort);
+    }
 
-        $modelObj = Injector::inst()->create($this->modelClass);
-        $searchableFields = $modelObj->searchableFields();
-        foreach ($this->searchParams as $key => $value) {
-            $key = str_replace('__', '.', $key ?? '');
-            if ($filter = $this->getFilter($key)) {
-                $filter->setModel($this->modelClass);
-                $filter->setValue($value);
-                if (!$filter->isEmpty()) {
-                    if (isset($searchableFields[$key]['match_any'])) {
-                        $searchFields = $searchableFields[$key]['match_any'];
-                        $filterClass = get_class($filter);
-                        $modifiers = $filter->getModifiers();
-                        $query = $query->alterDataQuery(function (DataQuery $dataQuery) use ($searchFields, $filterClass, $modifiers, $value) {
-                            $subGroup = $dataQuery->disjunctiveGroup();
-                            foreach ($searchFields as $matchField) {
-                                /** @var SearchFilter $filterClass */
-                                $filter = new $filterClass($matchField, $value, $modifiers);
-                                $filter->apply($subGroup);
-                            }
-                        });
-                    } else {
-                        $query = $query->alterDataQuery([$filter, 'apply']);
+    /**
+     * Use the global primary search for searching across multiple fields
+     *
+     * @param string|array $searchPhrase
+     */
+    protected function primaryFieldSearch(DataList $query, array $searchableFields, $searchPhrase): DataList
+    {
+        $context = $this;
+        $modelClass = $this->modelClass;
+        return $query->alterDataQuery(function (DataQuery $dataQuery) use ($context, $modelClass, $searchableFields, $searchPhrase) {
+            $formFields = $context->getSearchFields();
+            // Function which takes a search phrase or search term and searches for it across all searchable fields.
+            $searchAcrossFields = function ($searchPhrase, $subGroup) use ($context, $modelClass, $formFields, $searchableFields) {
```

Comment:
```suggestion
            $searchAcrossFields = function ($searchPhrase, $subGroup) use ($this, $modelClass, $formFields, $searchableFields) {
```

Reply:
Not a fan of putting anonymous functions to variables, private class methods read better IMO

Reply:
I agree, but this is in the context of an anonymous function already - I can't call a private method from within an anonymous function, which is why I went for a nested anonymous function instead. The alternative would be making it a public method which I don't think is appropriate.

Reply:
Moved to new private method